src/prompt.c

```diff
@@ -0,0 +1,68 @@
+#include "prompt.h"
+#include "stdio.h"
+
+t_prompt 	*prompt_create(const char *_username, const char *_hostname)
+{
+	t_prompt *prompt;
+
+	prompt = ft_calloc(sizeof(t_prompt), 1);
+	if (!prompt)
+		return (NULL);
+	prompt->username = ft_strdup(_username);
+	prompt->hostname = ft_strdup(_hostname);
+	prompt->buffer = ft_calloc(sizeof(char), BUFFER_SIZE);
+	if (!prompt->buffer || !prompt->username || !prompt->hostname)
+	{
+		prompt_destroy(prompt);
+		return (NULL);
+	}
+	/*prompt->current_path = directory_get_current_path();  -> first need to merge directory branch*/
+	prompt->error_code = 0;
+	prompt_init_buffer(prompt);
+	return (prompt);
+}
+
+void	prompt_init_buffer(t_prompt *_prompt)
+{
+	if (_prompt->error_code != 0)
+		snprintf(_prompt->buffer, ERROR_LENGTH, "%d", _prompt->error_code);
+	else
+		snprintf(_prompt->buffer, ERROR_LENGTH, "   ");
+	snprintf(_prompt->buffer, PROMPT_LENGTH, " %s@%s:[%s]>", _prompt->username, _prompt->hostname, _prompt->current_path);		
+}
```

Comment:
init buffer should malloc the buffer, then set all but the last byte which should be a NULLBYTE for protection to a space ' '.
then write the data that never changes into the buffer, like user@hostname
that's why you need to have indexes in the prompt, to know where to write stuff.

Reply:
you mean in my struct I should have : 
int error_index;
int username_index;   
int hostname_index;